examples/codenames/Codenames_RL.ipynb

```diff
@@ -0,0 +1,12525 @@
+        "        revealed = game_state.get_revealed_cards()\n",
+        "\n",
+        "        lines = []\n",
+        "        lines.append(f\"Number of words left for blue: {game_state.words_left['blue']}\")\n",
+        "        lines.append(f\"Number of words left for red: {game_state.words_left['red']}\")\n",
+        "        lines.append(\"UNREVEALED WORDS (no color info):\")\n",
+        "        for c in unrevealed:\n",
+        "            lines.append(f\"  {c.word}\")\n",
+        "        lines.append(\"REVEALED WORDS (with color):\")\n",
+        "        for c in revealed:\n",
+        "            lines.append(f\"  {c.word}:{c.color}\")\n",
+        "        return \"\\n\".join(lines)\n",
+        "\n",
+        "    async def guess_word(\n",
+        "        self,\n",
+        "        gs: GameState,\n",
+        "        clue_word: str,\n",
+        "        clue_number: int,\n",
+        "        game_events: str,\n",
+        "    ) -> str:\n",
+        "        if self.new_turn:\n",
+        "            self.new_turn = False\n",
+        "            if game_events:\n",
+        "                self.conversation.append({\"role\": \"user\", \"content\": game_events})\n",
+        "            prompt = OPERATIVE_USER_PROMPT.format(\n",
+        "                team=self.team,\n",
+        "                clue=clue_word,\n",
+        "                num=clue_number,\n",
+        "                board=self._board_summary(gs),\n",
+        "            )\n",
+        "            self.conversation.append({\"role\": \"user\", \"content\": prompt})\n",
+        "\n",
+        "        resp = await call_llm(\n",
+        "            self.model,\n",
+        "            self.conversation,\n",
+        "            self.temperature,\n",
+        "            self.max_tokens,\n",
+        "        )\n",
+        "        self.conversation_choices.append(resp.choices[0])\n",
+        "        guess = resp.choices[0].message.content.strip()  # type: ignore\n",
+        "        self.conversation.append({\"role\": \"assistant\", \"content\": guess})\n",
+        "        return guess\n",
+        "\n",
+        "    def start_new_turn(self):\n",
+        "        self.new_turn = True\n",
+        "\n",
+        "    def add_guess_outcome(self, outcome: str):\n",
+        "        self.conversation.append({\"role\": \"user\", \"content\": outcome})\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "x5zSJt3fwD0l"
+      },
+      "outputs": [],
+      "source": [
+        "# ───────────────────────  Gameplay  ──────────────────────────\n",
+        "async def run_game(\n",
+        "    gs: GameState,\n",
+        "    blue_spy: LLMSpymaster,\n",
+        "    blue_op: LLMOperative,\n",
+        "    red_spy: LLMSpymaster,\n",
+        "    red_op: LLMOperative,\n",
+        "    enforce_clue_rules: bool = True,\n",
```

Comment:
It's confusing to me to have `enforce_clue_rules` be both a function name and a boolean parameter to a function. Could we either rename this parameter or remove it entirely? Is there a circumstance in which we don't want to enforce clue rules?